### PR TITLE
[WIP] Move TBANS code to tasks_io (again). Add tasks_io endpoints for TBANS

### PR DESCRIPTION
**Pull request by ZachOrr**
This code shouldn't be in common really - so I'm moving it out, but doing it properly this time.

### CODE REVIEW

**Comment by fangeugene** [comment]
image

File: src/backend/tasks_io/models/notification_requests/fcm_request.py [hunk 1]
```diff
@@ -44,12 +52,12 @@ def __init__(self, app, notification, tokens=None, legacy_data_format=False):
...
         self.legacy_data_format = legacy_data_format
 
-    def __str__(self):
+    def __str__(self) -> str:
         return "FCMRequest(tokens={}, notification={})".format(
             self.tokens, str(self.notification)
         )
 
-    def send(self):
+    def send(self) -> messaging.BatchResponse:
```
Comment (ZachOrr) [inline src/backend/tasks_io/models/notification_requests/fcm_request.py:60]: - [ ] The return type here between `FCMRequest` and `WebhookRequest` is different. I'm surprised our linter doesn't catch this. Think about how to standardize this.

**Comment by ZachOrr** [comment]
This is reminding me of the cross-service request structure I built back in 2019 for this (and later removed) https://github.com/ZachOrr/the-blue-alliance/blob/ef991510a7f7e9682b24a64e2e508c85686d01fa/helpers/tbans_helper.py

iirc the removal was because of timeout issues across services? I need to think more about this. The cross-service situation is better than regular HTTP requests, since unlike the rest of the stuff that makes requests to `tasks-io`, these are making requests synchronously (usually) whereas other requests are leveraging deferred for cross-service execution.

**Comment by ZachOrr** [comment]
Alright for sure closing this out

**Pull request closed without merging**